fe/fe-core/src/main/java/org/apache/doris/catalog/Catalog.java

```diff
@@ -2463,10 +2463,12 @@ public void addFrontend(FrontendNodeType role, String host, int editLogPort) thr
 
             fe = new Frontend(role, nodeName, host, editLogPort);
             frontends.put(nodeName, fe);
+            BDBHA bdbha = (BDBHA) haProtocol;
             if (role == FrontendNodeType.FOLLOWER || role == FrontendNodeType.REPLICA) {
-                ((BDBHA) getHaProtocol()).addHelperSocket(host, editLogPort);
+                bdbha.addHelperSocket(host, editLogPort);
```

Comment:
Where does the conflict occur, and should we `removeConflictNodeIfExist` and then `addHelperSocket`?

Reply:
> Where does the conflict occur, and should we `removeConflictNodeIfExist` and then `addHelperSocket`?

help socket is used for self to connect with.  firstly add help socket, secondly we can communicate with cluster.